lib/api/model/events.dart

```diff
@@ -832,6 +834,71 @@ class UpdateMessageFlagsMessageDetail {
   Map<String, dynamic> toJson() => _$UpdateMessageFlagsMessageDetailToJson(this);
 }
 
+
+@JsonSerializable(fieldRename: FieldRename.snake)
+class TypingStatusEvent extends Event {
+  @override
+  @JsonKey(includeToJson: true)
+  String get type => 'typing';
+
+  final TypingStatusOp op;
+  final MessageType messageType;
+  final Typist sender;
+  final List<Typist>? recipients;
+  final int? streamId;
+  final String? topic;
+
+  TypingStatusEvent({
+    required super.id,
+    required this.op,
+    required this.messageType,
+    required this.sender,
+    required this.recipients,
+    required this.streamId,
+    required this.topic,
+  });
+
+  factory TypingStatusEvent.fromJson(Map<String, dynamic> json) {
+    final result = _$TypingStatusEventFromJson(json);
+    // Crunchy-shell validation
+    switch (result.messageType) {
+      case MessageType.stream:
+        result.streamId as int;
+        result.topic as String;
+      case MessageType.private:
+        result.recipients as List<Typist>;
+    }
+    return result;
+  }
+
+  @override
+  Map<String, dynamic> toJson() => _$TypingStatusEventToJson(this);
+}
+
+/// As in [TypingStatusEvent.op].
+@JsonEnum(fieldRename: FieldRename.snake)
+enum TypingStatusOp {
+  start,
+  stop
+}
+
+/// As in [TypingStatusEvent.sender] and [TypingStatusEvent.recipients].
+@JsonSerializable(fieldRename: FieldRename.snake)
+class Typist {
```

Comment:
This name "Typist" fits for the sender, but not for the recipients.

Even if it were just the sender, the disadvantage of this kind of name is that it doesn't clearly tie the class back to these events. This is basically a single-use-site helper for this event class, so it's good for the name to reflect that so it doesn't sound like something potentially more general.

A good boring name would then be `TypingEventUser`.